### PR TITLE
List metrics that relate to Pod security admission

### PR DESCRIPTION
**Pull request by ugur99**
Signed-off-by: Ugur Can Ozturk [ugurozturk918@gmail.com](mailto:ugurozturk918@gmail.com)

I added psa metrics to the document. They might be useful to monitor psa activities.

### CODE REVIEW

File: content/en/docs/concepts/security/pod-security-admission.md [hunk 1]
```diff
@@ -125,6 +125,14 @@ current policy level:
 - Valid updates to `.spec.activeDeadlineSeconds`
 - Valid updates to `.spec.tolerations`
 
+## Metrics
+You can look at the following Prometheus metrics exposed by kube-apiserver.
```
Comment (windsonsea) [inline content/en/docs/concepts/security/pod-security-admission.md:129]:
```suggestion
## Metrics

Here are the Prometheus metrics exposed by kube-apiserver:
```

File: content/en/docs/concepts/security/pod-security-admission.md [hunk 2]
```diff
@@ -125,6 +125,14 @@ current policy level:
...
+
+- `pod_security_errors_total`: metric contains the number of errors preventing normal evaluation. Non-fatal errors may result in the latest restricted profile being used for evaluation.
```
Comment (windsonsea) [inline content/en/docs/concepts/security/pod-security-admission.md:131]:
```suggestion
- `pod_security_errors_total`: This metric indicates the number of errors preventing normal evaluation.
  Non-fatal errors may result in the latest restricted profile being used for evaluation.
```

File: content/en/docs/concepts/security/pod-security-admission.md [hunk 3]
```diff
@@ -125,6 +125,14 @@ current policy level:
...
+- `pod_security_evaluations_total`: metric contains the number of policy evaluations that occurred, not counting ignored or exempt requests.
+  during exporting.
```
Comment (windsonsea) [inline content/en/docs/concepts/security/pod-security-admission.md:133]:
```suggestion
- `pod_security_evaluations_total`: This metric indicates the number of policy evaluations that have occurred,
  not counting ignored or exempt requests during exporting.
```

File: content/en/docs/concepts/security/pod-security-admission.md [hunk 4]
```diff
@@ -125,6 +125,14 @@ current policy level:
...
+- `pod_security_exemptions_total`: metric contains the number of exempt requests, not counting ignored or out of scope requests.
```
Comment (windsonsea) [inline content/en/docs/concepts/security/pod-security-admission.md:134]:
```suggestion
- `pod_security_exemptions_total`: This metric indicates the number of exempt requests, not counting ignored
  or out of scope requests.
```

**Review by windsonsea — COMMENTED** [review]
Revised some text for suggesion.

**Comment by ugur99** [comment]
> Revised some text for suggesion.

Thank you for review @windsonsea! I fixed them.

**Comment by sftim** [comment]
/retitle List metrics that relate to Pod security admission

**Comment by ugur99** [comment]
Is there anything you expect me to do about this issue? @sftim

**Comment by mengjiao-liu** [comment]
for tech review 
@tallclair @dgrisonnet

**Comment by ugur99** [comment]
Hey guys, sorry to bother you again; any updates about this one? @tallclair @dgrisonnet

**Review by dgrisonnet — APPROVED** [review]
I don't have any knowledge about these particular metrics, but having them here for discoverability with their help text sounds reasonable to me :+1: 

/lgtm

**Comment by reylejano** [comment]
I'll defer to SIG Auth for a technical review but the [PSA KEP readme](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement) lists the same three metrics listed in this PR:
- `pod_security_evaluations_total` : This metric will be added to track policy evaluations against pods and [templated pods](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement#podtemplate-resources). [Namespace evaluations](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement#namespace-policy-update-warnings) are not counted. The metric will only be incremented when the policy check is actually performed.
- `pod_security_exemptions_total` : This metric will be added to track requests that are considered exempt. Ignored resources and out of scope requests do not count towards the total.
- `pod_security_errors_total` : This metric will be added to track errors encountered during request evaluation.

/lgtm

**Review by tallclair — COMMENTED** [review]
/lgtm

**Comment by divya-mohan0209** [comment]
@ugur99 : Please could you address the nit? I'd be happy to get this merged right away once done :)

**Comment by ugur99** [comment]
> @ugur99 : Please could you address the nit? I'd be happy to get this merged right away once done :)

I'm a bit excited to contribute to the k8s website for the first time :) @divya-mohan0209

**Comment by sftim** [comment]
https://github.com/kubernetes/website/pull/41814#discussion_r1293784723 is only a nit

/approve

**Pull request merged**